Fix for Issue #3655: Update Linear Regression Sample in Python Bindings

MarkFischinger opened the pull request:
This PR addresses issue [#3655](https://github.com/mlpack/mlpack/issues/3655) reported by [tunglinwood](https://github.com/tunglinwood). The issue came from changes in the Linear Regression, Ridge Regression and Bayesian Linear Regression python bindings, which you can see in the [mlpack documentation](https://www.mlpack.org/doc/stable/python_documentation.html#class-linear_regression). The sample code provided earlier does not function as expected under these new changes.

**Modifications**
- Modified the correlation matrix calculation to focus only on numerical columns.
- Updated the Linear Regression & Ridge Regression & Bayesian Linear Regression model setup using the new constructor.

rcurtin commented on airbnb_price_prediction_regression/airbnb_price_prediction_regression_py.ipynb:
```diff
@@ -991,7 +991,7 @@
    "outputs": [
```
<pre class="ql-syntax" spellcheck="false">Line #1.    model = mlpack.LinearRegression(check_input_matrices=False, copy_all_inputs=False, lambda_=0.0, verbose=False)
</pre><p>Aren't these all the default parameters?  If so I would omit them just for the sake of cleanliness.  Users can find more details on options in the documentation, there's no need to add them all here (it could seem overwhelming).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlpack/examples/pull/224/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='93'/>

rcurtin replied:
<pre class="ql-syntax" spellcheck="false">Line #31.    print(f"Mean Absoulte Error: {np.mean(mae_score):.2f}")
</pre><p>I know this isn't what you intended to fix, but can you fix the typo on this line while you're at it? :)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlpack/examples/pull/224/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='93'/>

rcurtin replied:
<pre class="ql-syntax" spellcheck="false">Line #1.    model = mlpack.LinearRegression(check_input_matrices=False, copy_all_inputs=False, lambda_=0.1, verbose=False)
</pre><p>Same comments on this cell about default parameters (and also the 'absoulte' typo :)).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/mlpack/examples/pull/224/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='97'/>

rcurtin commented:
Thanks @MarkFischinger for looking into this!  The fixes look good to me, just some minor comments. :+1:

MarkFischinger commented:
Hi @rcurtin,
cleaned up the typos and simplified the initialization of the LinearRegression model. Also, I've updated the `df.corr()` part for better clarity :)
Initially, I thought it might be smart to include the default parameters, so the user knows what to expect from this class, but your point on the readability and simplicity outweighs.
